modules/cold-extraction/ColdDataRetriever.py

```diff
@@ -130,24 +130,40 @@ def initialize():
 with open(csv_file, newline='') as f:
     reader = csv.reader(f)
     next(f)
+
+    # Changed below part for finding missing csv entries and skipping them
     for row in reader:
+        row = [x.strip() for x in row]
+        #print(row)
         if (extraction_type == 'empi_date'):
-            patients.append(row[patient_index])
-            temp_date = row[date_index]
-            dt_stamp = datetime.datetime.strptime(temp_date, date_format)
-            date_str = dt_stamp.strftime('%Y%m%d')
-            dates.append(date_str)
-            length = len(patients)
+            if set(row).pop()=='':
```

Comment:
Also, this is checking for an empty entry. Please note that you should check only for the missing entry of the mandatory fields.

1) empi_date: make sure neither empi nor date is empty (there could be other fields that can be empty).
2) empi: make sure empi is not empty  (there could be other fields that can be empty).
3) accession: make sure accession is not empty  (there could be other fields that can be empty).
4) empi_accession: make sure neither empi nor accession is empty (there could be other fields that can be empty).

Reply:
Okay, understood my error
will work on it

Reply:
Made changes according to above given comments